core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -5897,6 +5897,13 @@ public final void updateRoleName(String roleName, String newRoleName) throws Use
         if (!isReadOnly() && writeGroupsEnabled) {
             try {
                 doUpdateRoleName(userStore.getDomainFreeName(), userStoreNew.getDomainFreeName());
+                if (hybridRoleManager.isGroupNameExistInHybridGroupRoleTable(roleName)) {
+                    String internalSystemRole = UserCoreConstants.INTERNAL_SYSTEM_ROLE_PREFIX +
+                            UserCoreUtil.extractDomainFromName(newRoleName)
+                                    .toLowerCase() + "_" + UserCoreUtil.removeDomainFromName(newRoleName);
+                    hybridRoleManager.updateGroupListOfHybridRole(internalSystemRole, new String[]{roleName},
```

Comment:
what happens if this internalSystemRole has more groups assigned? Those will get replaced and only newRoleName group will remain.

Reply:
those will remain as it is. Only the group entry which we are passing to updateGroupListOfHybridRole will get deleted and added back with new name.